refactor!: nuking remaining Aztec Node proxy methods from PXE

benesjan opened the pull request:
Continuation of the work from the PR down the sack and nuking the remaining Aztec Node proxy methods from PXE. These are the changes:
1. Dropepd `getCurrentBaseFee`,
2. moved `getTxReceipt` on `Wallet`,
3. drop `sendTx` and moved the implementation to `BaseWallet`,
4. dropped `getNodeInfo` (this was super ugly because we had essentially a copy of implementation from AztecNode in PXE).

The cluttering of wallet inputs is only temporary because PXE is supposed to be a lib used by the wallet and hence the wallet itself should "set it up".

This makes me realize that the PXEService naming is stale as it's transforming to not be a service. Will rename that once I nuke the PXE JSON RPC server.

benesjan commented:
* **#17109** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17109?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17109?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#17098** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17098?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

benesjan commented on yarn-project/aztec.js/src/contract/deploy_sent_tx.ts:
```diff
@@ -57,18 +56,13 @@ export class DeploySentTx<TContract extends Contract = Contract> extends SentTx
-    return { ...receipt, contract };
-  }
-
-  protected async getContractObject(wallet?: Wallet): Promise<TContract> {
```
This method was only used in the `wait` method so I decided to just merge them.

benesjan commented on yarn-project/aztec.js/src/contract/deploy_account_sent_tx.ts:
```diff
@@ -17,11 +16,11 @@ export type DeployAccountTxReceipt = FieldsOf<TxReceipt> & {
 export class DeployAccountSentTx extends SentTx {
   constructor(
-    pxeNodeOrWallet: Wallet | AztecNode | PXE,
+    wallet: Wallet,
```
This was always only used with a wallet so I constrained the type just to Wallet. This will help us later on figure out how it all is even used.

Thunkar replied:
this was used with a node by externals, can we at least keep the node?

benesjan replied:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/17109/commits/50117599bea73fed23be13474d67f9dcbfa8c6df

benesjan commented on yarn-project/aztec.js/src/wallet/wallet.ts:
```diff
@@ -48,8 +49,9 @@ export type ContractInstanceAndArtifact = Pick<Contract, 'artifact' | 'instance'
-  'getContractClassMetadata' | 'getContractMetadata' | 'getTxReceipt' | 'getPrivateEvents' | 'getPublicEvents'
+  'getContractClassMetadata' | 'getContractMetadata' | 'getPrivateEvents' | 'getPublicEvents'
 > & {
+  getTxReceipt(txHash: TxHash): Promise<TxReceipt>;
```
This is the method that triggered the slack discussion yesterday. Decided to keep it just on the `Wallet` for now as dropping this would result in a huge change.

benesjan commented on yarn-project/aztec.js/src/ethereum/l1_contracts.ts:
```diff
@@ -1,21 +0,0 @@
-
-import { createPXEClient } from '../rpc_clients/pxe_client.js';
-
-export const getL1ContractAddresses = async (url: string): Promise<L1ContractAddresses> => {
```
This was unused.

benesjan commented on yarn-project/aztec.js/src/wallet/account_manager.ts:
```diff
@@ -48,7 +49,14 @@ export class AccountManager {
 
-  static async create(wallet: Wallet, pxe: PXE, secretKey: Fr, accountContract: AccountContract, salt?: Salt) {
+  static async create(
+    wallet: Wallet,
+    pxe: PXE,
+    node: AztecNode,
```
Eventually, this should not accept pxe.

Thunkar replied:
The account manager is only ever created by wallets and with the exception of embedded wallets never shared outside of it, so it is (mostly) fine.

benesjan commented on yarn-project/aztec.js/src/wallet/base_wallet.ts:
```diff
@@ -292,7 +323,32 @@ export abstract class BaseWallet implements Wallet {
   }
-  getPublicEvents<T>(event: EventMetadataDefinition, from: number, limit: number): Promise<T[]> {
-    return this.pxe.getPublicEvents(event, from, limit);
+  async getPublicEvents<T>(eventMetadataDef: EventMetadataDefinition, from: number, limit: number): Promise<T[]> {
```
This will get dropped by Grego's PR so it's now inconsequential.

benesjan commented on yarn-project/cli/src/utils/aztec.test.ts:
```diff
@@ -1,31 +0,0 @@
-import { checkServerVersion } from './aztec.js';
-
-describe('aztec utils', () => {
-  describe('checkServerVersion', () => {
```
utils no longer used

benesjan commented on yarn-project/cli/src/utils/aztec.ts:
```diff
@@ -260,38 +259,3 @@ export function prettyPrintNargoToml(config: NoirPackageConfig): string {
- * @param pxe - PXE client.
- * @param expectedVersionRange - Expected version by CLI.
- */
-export async function checkServerVersion(pxe: PXE, expectedVersionRange: string) {
```
no longer used so dropped it

benesjan commented on yarn-project/end-to-end/src/devnet/e2e_smoke.test.ts:
```diff
@@ -123,18 +123,19 @@ describe('End-to-end tests for devnet', () => {
     } else if (PXE_URL) {
       logger.info(`Using PXE_URL: ${PXE_URL}`);
       pxe = createPXEClient(PXE_URL);
+      node = createAztecNodeClient(PXE_URL);
```
this is weird but both services share the URL here. This will get resolved with me dropping the PXE JSON RPC

Thunkar commented on yarn-project/aztec.js/src/wallet/account_manager.ts:
```diff
@@ -81,7 +89,7 @@ export class AccountManager {
    */
   public async getAccountInterface(): Promise<AccountInterface> {
-    const nodeInfo = await this.pxe.getNodeInfo();
+    const nodeInfo = await this.node.getNodeInfo();
```
Wallet now provides `getChainInfo` (which I believe it's what the account interface wants) so this file might not need changes

benesjan replied:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/17109/commits/7a405c36b514430e8249b1c59957c7642887debc

benesjan committed:
refactor!: nuking remaining Aztec Node proxy methods from PXE

Continuation of the work from the PR down the sack and nuking the remaining Aztec Node proxy methods from PXE. These are the changes:
1. Dropepd `getCurrentBaseFee`,
2. moved `getTxReceipt` on `Wallet`,
3. drop `sendTx` and moved the implementation to `BaseWallet`,
4. dropped `getNodeInfo` (this was super ugly because we had essentially a copy of implementation from AztecNode in PXE).

The cluttering of wallet inputs is only temporary because PXE is supposed to be a lib used by the wallet and hence the wallet itself should "set it up".

This makes me realize that the PXEService naming is stale as it's transforming to not be a service. Will rename that once I nuke the PXE JSON RPC server.
